### PR TITLE
Couple of fixes to issues I reported :)

### PR DESCRIPTION
**Pull request by macedot**
1. Change acServer running dir to its own path so checksum calculation for car and tracks should work;
2. Add parameters for:
- UDP parameters (needed for stracker/minorating)
- legal tyres, so one can restrict tyres usage
- wind parameters for weather
- fixed setup at entry list (stringl; will point to an existing .ini file at acServer/setups)

Not tested, but should build at least :)

### CODE REVIEW

**Comment by DeKugelschieber** [comment]
You modified the database schema, can you please update the db/schema.sql and add a migration script to upgrade from version 1.0.3? Just add a file db/mig_1.0.3_1.0.4.sql containing all instructions that need to be executed. The schema.sql should contain the complete database schema without data and comments.

Also, when I merge this it will break the master branch, since you haven't added your changes to the frontend. Are you familiar with VueJs/JavaScript? In that case can you add the new configurations to the public/app/pages/configuration.js? If not, I can take care of that.

Steps to build, run and (manually) test:
1. install Go 1.8
2. install node and npm
3. install node-sass (npm install -g node-sass)
4. install MySQL
5. download minvue and place it into public (https://github.com/DeKugelschieber/vuejs-minify), or just use the linux version (which is present already)
6. run minvue, node-sass and the server like in the bash scripts (/run, /public/build_app, /public/build_sass)

**Comment by macedot** [comment]
Done. Should I create a new pull request?

**Comment by DeKugelschieber** [comment]
No your commits get added to this pull request. I'll test the changes :)

File: public/app/pages/configuration.js [hunk 1]
```diff
@@ -75,6 +76,9 @@ Vue.component('Configuration', {
 			time: '08:00',
 			sun_angle: 32,
 			track: '',
+			legal_tyres: '',
+			udp_plugin_local_port: '',
```
Comment (venyii) [inline public/app/pages/configuration.js]: This should be `0` if we're following the AC Server Manager

File: public/app/pages/configuration.js [hunk 2]
```diff
@@ -393,6 +404,9 @@ Vue.component('Configuration', {
 				weather: this.weather,
 				track: this.track.name,
 				track_config: this.track.config,
+				legal_tyres: this.legal_tyres,
+				udp_plugin_local_port: this.udp_plugin_local_port,
```
Comment (DeKugelschieber) [inline public/app/pages/configuration.js]: Add parseInt() for numbers: udp_plugin_local_port: parseInt(this.udp_plugin_local_port),

File: public/app/pages/configuration.js [hunk 3]
```diff
@@ -189,6 +194,9 @@ Vue.component('Configuration', {
 			this.time = '08 =00';
```
Comment (DeKugelschieber) [inline public/app/pages/configuration.js]: Can you fix this to 08:00 please? It's not releated to your work, but when you're on this already :)

File: public/app/pages/configuration.js [hunk 4]
```diff
@@ -189,6 +194,9 @@ Vue.component('Configuration', {
...
 			this.sun_angle = 32;
 			this.track = '';
+			this.legal_tyres = '';
+			this.udp_plugin_local_port = '';
```
Comment (DeKugelschieber) [inline public/app/pages/configuration.js]: Change this to this.udp_plugin_local_port = 0;

File: public/app/pages/configuration.js (hunk 1 above)
Comment (DeKugelschieber) [inline public/app/pages/configuration.js]: Change value to 0, or another number default value (also in _reset function).

File: db/mig_1.0.3_1.0.4.sql [hunk 5]
```diff
@@ -0,0 +1,14 @@
+
+ALTER TABLE `cars`
+ADD COLUMN `fixed_setup` varchar(100) NOT NULL AFTER `position`;
+
+ALTER TABLE `configurations`
+ADD COLUMN `legal_tyres` varchar(100) NOT NULL AFTER `track_config`
```
Comment (DeKugelschieber) [inline db/mig_1.0.3_1.0.4.sql]: Missing ,

File: db/mig_1.0.3_1.0.4.sql [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+ADD COLUMN `udp_plugin_local_port` int(10) NOT NULL AFTER `legal_tyres`,
+ADD COLUMN `udp_plugin_address` varchar(100) NOT NULL AFTER `udp_plugin_address`;
```
Comment (DeKugelschieber) [inline db/mig_1.0.3_1.0.4.sql]: AFTER should refer to udp_plugin_local_port instead of udp_plugin_address.

**Comment by DeKugelschieber** [comment]
To do local testing, you can also use docker instead of setting up my complete dev environment. Install docker on your machine, setup a mysql database (https://hub.docker.com/_/mysql/) and build the image from within th acweb directory: docker build -t acweb . and then run it like on your server.

It really would help to make sure your changes actually compile and work propperly. I'm thinking about adding unit tests in a future version.

**Comment by macedot** [comment]
Did it in a hurry before leave to work. Will fix asap :)

File: public/app/pages/configuration.js (hunk 3 above)
Comment (macedot) [inline public/app/pages/configuration.js]: I'll change to 14:00 sun angle 16 coz it's the universal race time :)

File: public/app/pages/configuration.js (hunk 1 above)
Comment (macedot) [inline public/app/pages/configuration.js]: Agree.

File: public/app/pages/configuration.js (hunk 3 above)
Comment (DeKugelschieber) [inline public/app/pages/configuration.js]: Alright :)

**Comment by macedot** [comment]
Should be enough :)
My current internet isn't great so it will take some time to download docker image and test it.

File: public/app/pages/configuration.js [hunk 7]
```diff
@@ -72,9 +73,12 @@ Vue.component('Configuration', {
 			race_wait_time: 60,
 			race_extra_lap: false,
 			join_type: 1,
-			time: '08:00',
-			sun_angle: 32,
+			time: '14:00',
+			sun_angle: 16,
 			track: '',
+			legal_tyres: '',
```
Comment (DeKugelschieber) [inline public/app/pages/configuration.js]: legal_tyres has no field in the UI (html)

File: src/model/configuration.go [hunk 8]
```diff
@@ -67,19 +67,26 @@ type Configuration struct {
 	SunAngle              int       `json:"sun_angle"`
 	Track                 string    `json:"track"`
 	TrackConfig           string    `json:"track_config"`
+	LegalTyres            string    `json:"legal_tyres"`
+	UdpPluginPort         int       `json:"udp_plugin_local_port"`
+	UdpPluginAddr         string    `json:"udp_plugin_address"`
```
Comment (DeKugelschieber) [inline src/model/configuration.go]: Is UdpPluginAddr supposed to be a string? You used a checkbox in the frontend and another checkbox for UdpPluginPort, which is an int. I think you need to change them to number fields in the UI (and add parseInt() if you haven't done that yet).

At the moment I get a marshalling error: `json: cannot unmarshal bool into Go struct field Configuration.udp_plugin_address of type string`

Comment (macedot) [inline src/model/configuration.go]: It should be a string. will fix it.

**Comment by macedot** [comment]
New commits for #10, #12 and #13 .

**Comment by DeKugelschieber** [comment]
Add the new table columns to the migration script please.

File: src/model/configuration.go [hunk 9]
```diff
@@ -14,6 +14,9 @@ type Configuration struct {
 	AdminPwd              string    `json:"admin_pwd"`
 	PickupMode            bool      `json:"pickup_mode"`
 	LockEntryList         bool      `json:"lock_entry_list"`
+	RacePitWindowStart    int       `json:"race_pit_window_start"`
```
Comment (DeKugelschieber) [inline src/model/configuration.go]: Move this after UdpPluginAddr to match database table order.

File: src/model/configuration.go [hunk 10]
```diff
@@ -145,6 +156,9 @@ func (m *Configuration) saveConfiguration(tx *sql.Tx) error {
 			admin_pwd,
 			pickup_mode,
 			lock_entry_list,
+			race_pit_window_start,
```
Comment (DeKugelschieber) [inline src/model/configuration.go]: Move this after udp_plugin_address to match database table order and adjust parameters.

File: src/model/configuration.go [hunk 11]
```diff
@@ -282,6 +306,9 @@ func (m *Configuration) saveConfiguration(tx *sql.Tx) error {
 		admin_pwd = ?,
 		pickup_mode = ?,
 		lock_entry_list = ?,
+		race_pit_window_start = ?,
```
Comment (DeKugelschieber) [inline src/model/configuration.go]: Move this after udp_plugin_address to match database table order and adjust parameters.

File: src/model/configuration.go [hunk 12]
```diff
@@ -601,6 +660,9 @@ func scanConfiguration(row util.RowScanner) (*Configuration, error) {
 		&config.AdminPwd,
 		&config.PickupMode,
 		&config.LockEntryList,
+		&config.RacePitWindowStart,
```
Comment (DeKugelschieber) [inline src/model/configuration.go]: Move this after UdpPluginAddr to match database table order.

**Pull request merged**